nvflare/job_config/fed_job_config.py

```diff
@@ -0,0 +1,321 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import builtins
+import inspect
+import json
+import os
+import shutil
+from enum import Enum
+from typing import Dict
+
+from nvflare import SimulatorRunner
+from nvflare.job_config.fed_app_config import FedAppConfig
+from nvflare.private.fed.app.fl_conf import FL_PACKAGES
+
+CONFIG = "config"
+CUSTOM = "custom"
+FED_SERVER_JSON = "config_fed_server.json"
+FED_CLIENT_JSON = "config_fed_client.json"
+META_JSON = "meta.json"
+
+
+class FedJobConfig:
+    """FedJobConfig represents the job in the NVFlare."""
+
+    def __init__(self, job_name, min_clients, mandatory_clients=None) -> None:
+        """FedJobConfig uses the job_name,  min_clients and optional mandatory_clients to create the object.
+        It also provides the method to add in the FedApp, the deployment map of the FedApp and participants,
+        and the resource _spec requirements of the participants if needed.
+
+        Args:
+            job_name: the name of the NVFlare job
+            min_clients: the minimum number of clients for the job
+            mandatory_clients: mandatory clients to run the job (optional)
+        """
+        super().__init__()
+
+        self.job_name = job_name
+        self.min_clients = min_clients
+        self.mandatory_clients = mandatory_clients
+
+        self.fed_apps: Dict[str, FedAppConfig] = {}
+        self.deploy_map: Dict[str, str] = {}
+        self.resource_specs: Dict[str, Dict] = {}
+
+        self.custom_modules = []
+
+    def add_fed_app(self, app_name: str, fed_app: FedAppConfig):
+        if not isinstance(fed_app, FedAppConfig):
+            raise RuntimeError(f"server_app must be type of FedAppConfig, but got {fed_app.__class__}")
+
+        self.fed_apps[app_name] = fed_app
+
+    def set_site_app(self, site_name: str, app_name: str):
+        if app_name not in self.fed_apps.keys():
+            raise RuntimeError(f"fed_app {app_name} does not exist.")
+
+        self.deploy_map[site_name] = app_name
+
+    def add_resource_spec(self, site_name: str, resource_spec: Dict):
+        if site_name in self.resource_specs.keys():
+            raise RuntimeError(f"{site_name} resource specs already exist.")
+        if not isinstance(resource_spec, dict):
+            raise RuntimeError(f"resource_spec must be a dict. But got: {resource_spec.__class__}")
+
+        self.resource_specs[site_name] = resource_spec
+
+    def _generate_meta(self, job_dir):
+        """generate the job meta.json
+
+        Returns:
+
+        """
+        meta_file = os.path.join(job_dir, META_JSON)
+        meta_json = {
+            "name": self.job_name,
```

Comment:
We probably should use some constant variable for these key names

Reply:
We can combine this constant with the job scheduler meta.json parsing in the future.